pkg/operator/ceph/cluster/controller.go

```diff
@@ -471,6 +478,7 @@ func (c *ClusterController) csiVolumesAllowForDeletion(cluster *cephv1.CephClust
 		return nil
 	}
 
+	c.recorder.Event(cluster, "Warning", "FailedUninstall", fmt.Sprintf("waiting for csi volume attachments in cluster %q to be cleaned up", cluster.Namespace))
```

Comment:
It'd be nice to somehow wrap the error in the recorder at the top level to avoid having a new line before each error return. So potentially, we can end up with a single line, perhaps around here: 

https://github.com/rook/rook/blob/b4c55766e4fba46a70a3bc89e8120fa1e7bd6c67/pkg/operator/ceph/cluster/controller.go#L223

Reply:
Also, per my understanding, if this error occurs, the reconciler will be requeued with opcontroller.WaitForRequeueIfFinalizerBlocked (10 seconds). This means that the event will fire every 10 seconds. This won't clutter the describe, but it might spam the APIServer.

Reply:
> It'd be nice to somehow wrap the error in the recorder at the top level to avoid having a new line before each error return. So potentially, we can end up with a single line, perhaps around here:
> 
> https://github.com/rook/rook/blob/b4c55766e4fba46a70a3bc89e8120fa1e7bd6c67/pkg/operator/ceph/cluster/controller.go#L223

@leseb We can not do here as of know as we do not have a `cephCluster := &cephv1.CephCluster{}` object in this func, But we can do in the `reconcile` func or the other way around is that we need to get `cephCluster := &cephv1.CephCluster{}` in the `Reconcile` func as well. Let me know what do you think about this.

Reply:
Yes, that sounds fine, the goal is to have a few repetitive instances of the recorder as possible. Thanks

Reply:
@rohantmp Done